src/mlpack/methods/ann/layer/grouped_convolution_impl.hpp

```diff
@@ -0,0 +1,690 @@
+    padding(std::move(other.padding)),
+    paddingType(std::move(other.paddingType)),
+    inMaps(std::move(other.inMaps)),
+    higherInDimensions(std::move(other.higherInDimensions))
+{
+  // Nothing to do.
+}
+
+template<
+    typename ForwardConvolutionRule,
+    typename BackwardConvolutionRule,
+    typename GradientConvolutionRule,
+    typename MatType
+>
+GroupedConvolutionType<
+    ForwardConvolutionRule,
+    BackwardConvolutionRule,
+    GradientConvolutionRule,
+    MatType
+>&
+GroupedConvolutionType<
+    ForwardConvolutionRule,
+    BackwardConvolutionRule,
+    GradientConvolutionRule,
+    MatType
+>::operator=(const GroupedConvolutionType& other)
+{
+  if (&other != this)
+  {
+    Layer<MatType>::operator=(other);
+    maps = other.maps;
+    kernelWidth = other.kernelWidth;
+    kernelHeight = other.kernelHeight;
+    groups = other.groups;
+    strideWidth = other.strideWidth;
+    strideHeight = other.strideHeight;
+    padWLeft = other.padWLeft;
+    padWRight = other.padWRight;
+    padHBottom = other.padHBottom;
+    padHTop = other.padHTop;
+    useBias = other.useBias;
+    padding = other.padding;
+    paddingType = other.paddingType;
+    inMaps = other.inMaps;
+    higherInDimensions = other.higherInDimensions;
+  }
+
+  return *this;
+}
+
+template<
+    typename ForwardConvolutionRule,
+    typename BackwardConvolutionRule,
+    typename GradientConvolutionRule,
+    typename MatType
+>
+GroupedConvolutionType<
+    ForwardConvolutionRule,
+    BackwardConvolutionRule,
+    GradientConvolutionRule,
+    MatType
+>&
+GroupedConvolutionType<
+    ForwardConvolutionRule,
+    BackwardConvolutionRule,
+    GradientConvolutionRule,
+    MatType
+>::operator=(GroupedConvolutionType&& other)
+{
+  if (&other != this)
+  {
+    Layer<MatType>::operator=(std::move(other));
+    maps = std::move(other.maps);
+    kernelWidth = std::move(other.kernelWidth);
+    kernelHeight = std::move(other.kernelHeight);
+    groups = std::move(other.groups);
+    strideWidth = std::move(other.strideWidth);
+    strideHeight = std::move(other.strideHeight);
+    padWLeft = std::move(other.padWLeft);
+    padWRight = std::move(other.padWRight);
+    padHBottom = std::move(other.padHBottom);
+    padHTop = std::move(other.padHTop);
+    useBias = std::move(other.useBias);
+    padding = std::move(other.padding);
+    paddingType = std::move(other.paddingType);
+    inMaps = std::move(other.inMaps);
+    higherInDimensions = std::move(other.higherInDimensions);
+  }
+
+  return *this;
+}
+
+template<
+    typename ForwardConvolutionRule,
+    typename BackwardConvolutionRule,
+    typename GradientConvolutionRule,
+    typename MatType
+>
+void GroupedConvolutionType<
+    ForwardConvolutionRule,
+    BackwardConvolutionRule,
+    GradientConvolutionRule,
+    MatType
+>::SetWeights(typename MatType::elem_type* weightPtr)
+{
+  MakeAlias(weight, weightPtr, kernelWidth, kernelHeight, 
+      (maps * inMaps) / groups);
+    if (useBias)
```

Comment:
```suggestion
  if (useBias)
```
Oops, accidental extra indentation.